client/state/media/reducer.js

```diff
@@ -289,12 +289,16 @@ export const selectedItems = withoutPersistence( ( state = {}, action ) => {
 			};
 		}
 		case MEDIA_ITEM_REQUEST_SUCCESS: {
-			const { mediaId: transientMediaId, siteId } = action;
+			const { mediaId: requestedMediaId, siteId } = action;
 			const media = state[ siteId ] ?? [];
 
+			const isTransient = typeof requestedMediaId === 'string';
```

Comment:
I'm not sure if that's a valid assumption to be honest; are non-transient media item IDs always numbers?

Reply:
> while non-transient ones are always ~strings~ numbers

Just to clarify, this is what you meant, right?

Reply:
Of course, that's what I meant 😅

For the protocol, I find it weird that we use multiple different prefixes. It's all media after all that we upload there, so I think ideally we should use the same `media-` prefix everywhere. Any idea the separate prefix has any utility function?